azblob/zc_policy_unique_request_id.go

```diff
@@ -14,9 +14,20 @@ func NewUniqueRequestIDPolicyFactory() pipeline.Factory {
 		return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
 			id := request.Header.Get(xMsClientRequestID)
 			if id == "" { // Add a unique request ID if the caller didn't specify one already
-				request.Header.Set(xMsClientRequestID, newUUID().String())
+				id = newUUID().String()
+				request.Header.Set(xMsClientRequestID, id)
 			}
-			return next.Do(ctx, request)
+
+			resp, err := next.Do(ctx, request)
+
+			val := resp.Response().Header.Values(xMsClientRequestID)
+			if len(val) > 0 {
+				if val[0] != id {
+					err = errors.New("client Request ID from request and response does not match")
+				}
```

Comment:
I would prefer wrapping the old error in this one, if the old error isn't nil. [You should check out this news article on the Go website.](https://blog.golang.org/go1.13-errors)

Reply:
We don't have an error to wrap though, we should only check the client ID and throw the new error if there wasn't already an error.